### PR TITLE
For managed nodegroups, propagate labels and taints as tags to ASG

### LINKED ISSUE #5575: [Story] Propagate labels and taints to ASG with `propagateASGTags` for managed nodegroups
### Description

This is a follow-up to https://github.com/weaveworks/eksctl/issues/5420 and https://github.com/weaveworks/eksctl/issues/3961. 

The aim is to update `propagateASGTags` so that its behaviour with managed nodegroups is the same as with unmanaged nodegroups e.g. labels and taints should also be propagated as ASG tags.

### Scope

- This is for managed nodegroups to propagate labels+taints as ASG tags when `propagateASGTags` is set to `true`.
 
### User story

As a eksctl user, so that I don't manually convert managed nodegroup's labels and taints as tags for ASG, I want them to be converted into tags and propagated automatically.

### PR DESCRIPTION
**Pull request by TiberiuGC**
### Description

As a eksctl user, so that I don't manually convert managed nodegroup's labels and taints as tags for ASG, I want them to be converted into tags and propagated automatically.
Closes #5575

### Manual testing details
Used following config (notice that `tags` are commented)
```
managedNodeGroups:
  - name: mng2
    labels:
      my-cool-label: pizza
    taints:
      - key: feaster
        value: "true"
        effect: NoSchedule
    propagateASGTags: true
#    tags:
#      [k8s.io/cluster-autoscaler/node-template/label/my-cool-label](http://k8s.io/cluster-autoscaler/node-template/label/my-cool-label): pizza
#      [k8s.io/cluster-autoscaler/node-template/taint/feaster](http://k8s.io/cluster-autoscaler/node-template/taint/feaster): "true:NoSchedule"
```

Checking AWS Console for the created ASG for the _managed_ nodegroup, we see the labels and taints being automatically propagated as tags 
image
image

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [ ] Backfilled missing tests for code in same general area :tada:
- [ ] Refactored something and made the world a better place :star2:

### CODE REVIEW

File: pkg/cfn/manager/nodegroup.go [hunk 1]
```diff
@@ -99,7 +104,39 @@ func (c *StackCollection) propagateManagedNodeGroupTagsToASGTask(ctx context.Con
...
+	// add nodegroup tags
+	for k, v := range ng.Tags {
+		tags[k] = v
+	}
+
+	return c.PropagateManagedNodeGroupTagsToASG(ng.Name, tags, asgNames, errorCh)
+}
+
+// convertLabelsAndTaintsIntoTags so that they can be propageted automatically to ASG afterwards
+func convertLabelsAndTaintsIntoTags(ng *api.ManagedNodeGroup) (map[string]string, error) {
```
Comment (Himangini) [inline pkg/cfn/manager/nodegroup.go:123]: can we add tests for this?

Comment (TiberiuGC) [reply pkg/cfn/manager/nodegroup.go:124]: 17ba36563440eb2e36085ead9ac7e8580b98f6a4

File: pkg/cfn/manager/nodegroup.go [hunk 2]
```diff
@@ -99,7 +105,39 @@ func (c *StackCollection) propagateManagedNodeGroupTagsToASGTask(ctx context.Con
...
+
+	// labels
+	for k, v := range ng.Labels {
+		result[labelsPrefix+k] = v
+	}
+
+	// taints
+	for _, taint := range ng.Taints {
+		if _, ok := ng.Labels[taint.Key]; ok {
+			return nil, fmt.Errorf("duplicate key found for taints and labels with taint key=value: %s=%s, and label: %s=%s", taint.Key, taint.Value, taint.Key, ng.Labels[taint.Key])
```
Comment (Himangini) [inline pkg/cfn/manager/nodegroup.go:135]: it'll be good to add a test for this error as well, wdyt? 💡

File: pkg/cfn/manager/nodegroup_test.go [hunk 3]
```diff
@@ -163,4 +165,60 @@ var _ = Describe("StackCollection NodeGroup", func() {
 			})
 		})
 	})
+
+	Describe("propagateManagedNodeGroupTagsToASGTask", func() {
```
Comment (Himangini) [inline pkg/cfn/manager/nodegroup_test.go:169]:
```suggestion
	Describe("propagate ManagedNodeGroup tags to ASG", func() {
```

File: pkg/cfn/manager/nodegroup.go (hunk 2 above)
Comment (TiberiuGC) [reply pkg/cfn/manager/nodegroup.go:135]: 74cde341357fe3e2a32f2ebc5e757c45f37d128e

**Review by Himangini — APPROVED** [review]
LGTM 👍🏻

**Pull request merged**